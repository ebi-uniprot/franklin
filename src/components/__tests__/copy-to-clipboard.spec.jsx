import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CopyToClipboard from '../copy-to-clipboard';

jest.mock('copy-to-clipboard', () => {
  return jest.fn();
});

describe('CopyToClipboard component', () => {
  let rendered;
  const toCopy = 'foo bar baz';
  beforeEach(() => {
    rendered = render(
      <CopyToClipboard
        toCopy={toCopy}
        beforeCopy="Copy this"
        afterCopy="Copied it"
      />
    );
  });

  it('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  it('should change button text when after copy', () => {
    const { getByText } = rendered;
    const beforeCopyButton = getByText('Copy this');
    fireEvent.click(beforeCopyButton);
    const afterCopyButton = getByText('Copied it');
    expect(afterCopyButton).toBeTruthy();
  });
});
