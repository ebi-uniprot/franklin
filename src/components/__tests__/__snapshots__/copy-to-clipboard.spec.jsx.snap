// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyToClipboard component should render 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  >
    Copy this
  </button>
</DocumentFragment>
`;
